dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/SubscribeToEventTest.war/src/io/openliberty/org/apache/faces40/fat/subscribe/SubscribeTest.java

```diff
@@ -0,0 +1,127 @@
+                try {
+                    testSubscribeToEvent(cc, event, myListener);
+                } catch (AssertionError ae) {
+                    throw new AssertionError("Failed testSubscribeToEvent after " + i + " attempt(s) with message: " + ae.getLocalizedMessage());
+                }
+            }
+
+            //UIComponent#unsubscribeToEvent should remove listeners
+            testUnsubscribeToEvent(cc, event, new MyTestListener());
+        } catch (AssertionError ae) {
+            return Boolean.toString(false) + " - " + ae.getLocalizedMessage();
+        }
+
+        return Boolean.toString(true);
+    }
+
+    /**
+     * Ensure UIComponent has no system event listeners for a given event and does not return null.
+     *
+     * @param cc - the UI Component
+     * @param event - the SystemEvent class
+     * @throws AssertionError if test fails
+     */
+    private void testGetListenersForEventClass(UIComponent cc, Class<? extends SystemEvent> event) throws AssertionError {
+        List<SystemEventListener> subscribeList = cc.getListenersForEventClass(event);
+        if (subscribeList == null || !subscribeList.isEmpty()) {
+            throw new AssertionError("testGetListenersForEventClass subscribeList was either null, or not empty: " + subscribeList);
+        }
+    }
+
+    /**
+     * Ensure UIComponent checks to make sure listener does not duplicate install.
+     *
+     * @param cc - the UIComponent object
+     * @param event - the SystemEvent class
+     * @param listener - the ComponentSystemEventListener object
+     * @throws AssertionError if test fails
+     */
+    private void testSubscribeToEvent(UIComponent cc, Class<? extends SystemEvent> event, ComponentSystemEventListener listener) throws AssertionError {
+        cc.subscribeToEvent(event, listener);
+        List<SystemEventListener> subscribeList = cc.getListenersForEventClass(event);
+        if (subscribeList == null || subscribeList.size() != 1) {
+            throw new AssertionError("testSubscribeToEvent subscribeList was either null, or contained more than one listener: " + subscribeList);
+        }
+    }
+
+    /**
+     * For completeness check to make sure listener is unsubscribed successfully.
+     *
+     * @param cc - the UIComponent object
+     * @param event - the SystemEvent class
+     * @param listener - the ComponentSystemEventListener object
+     * @throws AssertionError if test fails
+     */
+    private void testUnsubscribeToEvent(UIComponent cc, Class<? extends SystemEvent> event, ComponentSystemEventListener listener) throws AssertionError {
+        cc.unsubscribeFromEvent(event, listener);
+        List<SystemEventListener> subscribeList = cc.getListenersForEventClass(event);
+        if (subscribeList == null || !subscribeList.isEmpty()) {
+            throw new AssertionError("testGetListenersForEventClass subscribeList was either null, or not empty: " + subscribeList);
```

Comment:
Should be `testUnsubscribeToEvent` instead of testGetListenersForEventClass in the message.

Reply:
Oh, I see why it failed.  You called unsubscribeFromEvent which removed the only event in the list., So the check should look for null / empty.